spring-kafka/src/main/java/org/springframework/kafka/aot/KafkaRuntimeHintsRegistrar.java

```diff
@@ -0,0 +1,253 @@
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.listener.ConsumerProperties;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.support.LoggingProducerListener;
+import org.springframework.kafka.support.ProducerListener;
+import org.springframework.kafka.support.serializer.DelegatingByTopicDeserializer;
+import org.springframework.kafka.support.serializer.DelegatingByTypeSerializer;
+import org.springframework.kafka.support.serializer.DelegatingDeserializer;
+import org.springframework.kafka.support.serializer.DelegatingSerializer;
+import org.springframework.kafka.support.serializer.ErrorHandlingDeserializer;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+import org.springframework.kafka.support.serializer.ParseStringDeserializer;
+import org.springframework.kafka.support.serializer.StringOrBytesSerializer;
+import org.springframework.kafka.support.serializer.ToStringSerializer;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.util.ClassUtils;
+
+/**
+ * {@link RuntimeHintsRegistrar} for Spring for Apache Kafka.
+ *
+ * @author Gary Russell
+ * @since 3.0
+ *
+ */
+public class KafkaRuntimeHintsRegistrar implements RuntimeHintsRegistrar {
+
+	@Override
+	public void registerHints(RuntimeHints hints, @Nullable ClassLoader classLoader) {
+		ReflectionHints reflectionHints = hints.reflection();
+		Stream.of(
+					PartitionOffset.class,
+					TopicPartition.class,
+					ConsumerProperties.class,
+					ContainerProperties.class,
+					ProducerListener.class,
+					KafkaListener.class,
+					EnableKafka.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_METHODS)));
+
+		Stream.of(
+					Message.class,
+					ImplicitLinkedHashCollection.Element.class,
+					KafkaListener.class,
+					MessageMapping.class,
+					KafkaListeners.class,
+					NewTopic.class,
+					AbstractKafkaListenerContainerFactory.class,
+					ConcurrentKafkaListenerContainerFactory.class,
+					KafkaListenerContainerFactory.class,
+					KafkaListenerEndpointRegistry.class,
+					DefaultKafkaConsumerFactory.class,
+					DefaultKafkaProducerFactory.class,
+					KafkaAdmin.class,
+					KafkaOperations.class,
+					KafkaResourceFactory.class,
+					KafkaTemplate.class,
+					ProducerFactory.class,
+					KafkaOperations.class,
+					ConsumerFactory.class,
+					LoggingProducerListener.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_CONSTRUCTORS,
+								MemberCategory.INVOKE_DECLARED_METHODS,
+								MemberCategory.INTROSPECT_PUBLIC_METHODS)));
+
+		Stream.of(
+					Message.class,
```

Comment:
Looks like you have duplicated this type.
I also wonder if we really use it via reflection at all...
In SI I registered it only for serialization.

Reply:
Again, I just copied what was in spring-native, for now.